### PR TITLE
Create base component and base decorator classes

### PR DESCRIPTION
**Pull request by Ellon-M**
## Changes

 - Added `nameable.rb` which contains the base component `Nameable` class.
 - Created a base decorator class `Decorator` which inherits from `Nameable`.
 - Created classes `CapitalizeDecorator` and `TrimmerDecorator` which extend the base `Decorator` class to add specialized functionality.

### CODE REVIEW

File: app/nameable.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  def correct_name
+    @correct = @nameable.correct_name
+    @correct.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @correct = @nameable.correct_name
+    @correct.length > 10 ? "#{@correct[0...10]}..." : @correct
```
Comment (julie-ify) [inline app/nameable.rb]: - [ ] Kindly remove these three dots `...` as it is unnecessary.

Comment (julie-ify) [inline app/nameable.rb]: - [ ] **[OPTIONAL]** I would recommend you move these classes to different files.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Ellon-M,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Ellon-M👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Well-implemented decorate pattern. ✔️
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**